overlord/devicestate/firstboot.go

```diff
@@ -189,7 +189,9 @@ func populateStateFromSeedImpl(st *state.State) ([]*state.TaskSet, error) {
 
 	// chain together configuring core, kernel, and gadget after
 	// installing them so that defaults are availabble from gadget
-	configTss[0].WaitAll(tsAll[last])
+	if len(tsAll) > last {
```

Comment:
I would have used len(configTss) > 0  around the full 3 lines, but no strong opinion

Reply:
Indeed, I think this is a bit easier to read.